common/embed.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *  nextpnr -- Next Generation Place and Route
+ *
+ *  Copyright (C) 2020  whitequark <whitequark@whitequark.org>
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ *  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ *  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#ifndef EMBED_H
+#define EMBED_H
+
+#include "nextpnr.h"
+NEXTPNR_NAMESPACE_BEGIN
+
+#if !defined(EXTERNAL_CHIPDB_ROOT) && !defined(WIN32)
+
+struct EmbeddedFile
+{
+    static std::map<std::string, const void *> all;
+
+    EmbeddedFile(const std::string &filename, const void *content) {
```

Comment:
Because C++ is weird, it is guaranteed that `all` will be initialised before the EmbeddedFile constructor runs, right?`

Reply:
Agh, you're right: this isn't guaranteed AFAIK, it depends on linking order. Let's make it a linked list? Then it doesn't matter which order this is initialized in.

Reply:
Done with a linked list